core/src/main/java/io/grpc/PickFirstBalancerFactory.java

```diff
@@ -92,9 +93,13 @@ public void handleNameResolutionError(Status error) {
     @Override
     public void handleSubchannelState(Subchannel subchannel, ConnectivityStateInfo stateInfo) {
       ConnectivityState currentState = stateInfo.getState();
-      if (subchannel != this.subchannel || currentState == SHUTDOWN) {
+      if (subchannel != this.subchannel) {
         return;
       }
+      if (currentState == SHUTDOWN) {
+        // Either IDLE_TIMEOUT or channel is shutdown. Won't have any effect for the latter case.
+        helper.updateBalancingState(IDLE, new Picker(PickResult.withNoResult()));
```

Comment:
IDLE_TIMEOUT is completely a channel impl logic. Even if we wanted to expose it to the application, it should be channel impl that would do that, i.e., transit to `IDLE` when `IdleModeTimer` expires and ignore updates from balancer if it's not the current one (which channel impl should be probably doing in the first place).

The balancer should not assume the channel's state. It should only provide state from its own perspective.